nelc/event-routing-backends
# feat: implement XAPI openedx filters

**Description:** 

This adds openedx filters to the requirements list a includes a basic implementation that allows to  modify or filter the output of the get_actos and get_objects event methods for the XAPI processor.

**Merge deadline:** Tomorrow 

**Testing instructions:**
This works like another openedx-filter https://github.com/openedx/openedx-filters/

1. Define your pipeline, follow the openedx-filter instructions,finally the logic is up to you.A basic implementation could be the following
```
import logging
from openedx_filters import PipelineStep

LOGGER = logging.getLogger(__name__)

class Something(PipelineStep):

    def run_filter(self, result):
        LOGGER.info("I'm very happy with my first pipeline")
        return {
            "result": result
        }
```

2. include the configuration in your settings

```
OPEN_EDX_FILTERS_CONFIG = {
    "event_routing_backends.processors.xapi.transformer.xapi_transformer.get_actor": {
        "pipeline": ["path.to.the.class.Something"],
        "fail_silently": False,
    }
}
```
3. Raise any event and check that the pipeline logic was executed
